### PR TITLE
When merging from fork the target is the fork's parent

### PR DESCRIPTION
**Pull request by majamassarini**
The PullRequest target project should be the parent of the forked project when merging from fork.

Fixes packit/hardly#63

### CODE REVIEW

**Review by mfocko — APPROVED** [review]
LGTM :) could you please update the same thing in the GitHub and GitLab implementation?

As I have mentioned in the mail, it should be those lines:
GitLab: https://github.com/packit/ogr/blob/3fd707590a3545109f3ce10e14e96c89f200a9b9/ogr/services/gitlab/pull_request.py#L197
GitHub: https://github.com/packit/ogr/blob/3fd707590a3545109f3ce10e14e96c89f200a9b9/ogr/services/github/pull_request.py#L165

Ideally I would set a variable with `target_project` ot be explicit about it when passing the project to the constructor

**Comment by majamassarini** [comment]
> LGTM :) could you please update the same thing in the GitHub and GitLab implementation?

Yes, sure. I think the hardest thing is to build a test because I have no way to trigger the error in gitlab or github. I'll try.

**Comment by mfocko** [comment]
> > LGTM :) could you please update the same thing in the GitHub and GitLab implementation?
> 
> Yes, sure. I think the hardest thing is to build a test because I have no way to trigger the error in gitlab or github. I'll try.

If you want to have failed tests, it should be enough to:
1. fork some testing repository
2. create PR by calling `.create_pr` on your fork
3. enjoy red tests, cause `target_project` should be your fork instead of the upstream

(at least that's the way you can reproduce it for GitHub and GitLab since you have already fixed the Pagure)

**Comment by majamassarini** [comment]
> > Yes, sure. I think the hardest thing is to build a test because I have no way to trigger the error in gitlab or github. I'll try.
> 
> If you want to have failed tests, it should be enough to:
> 
> 1. fork some testing repository
> 2. create PR by calling `.create_pr` on your fork
> 3. enjoy red tests, cause `target_project` should be your fork instead of the upstream
> 
> (at least that's the way you can reproduce it for GitHub and GitLab since you have already fixed the Pagure)

Nice, Thanks!

File: tests/integration/github/test_pull_requests.py [hunk 1]
```diff
@@ -87,6 +87,7 @@ def test_pr_create_upstream_upstream(self):
 
         assert pr_upstream_upstream.title == "test: upstream <- upstream"
         assert pr_upstream_upstream.status == PRStatus.open
+        assert pr_upstream_upstream.target_project.is_fork == False
```
Comment (mfocko) [inline tests/integration/github/test_pull_requests.py:90]:
```suggestion
        assert not pr_upstream_upstream.target_project.is_fork
```

also flake from pre-commit doesn't like it the way it is now ;)

**Review by mfocko — APPROVED** [review]
LGTM :)

File: tests/integration/github/test_pull_requests.py (hunk 1 above)
Comment (majamassarini) [reply tests/integration/github/test_pull_requests.py:90]: 😅

**Review by mfocko — APPROVED** [review]

**Comment by jpopelka** [comment]
Tested on Gitlab (staging CentOS Stream source-git/dist-git) and looks OK  :rocket:

**Pull request merged**